src/lib.js

```diff
@@ -677,6 +676,40 @@ var program,
       }
     },
 
+    /**
+     * Identifies whether the input data string is a graphql query or not
+     *
+     * @param {string} dataString - Input data string to check if it is a graphql query
+     * @returns {Object} - { result: true, graphql: {Object} } if dataString is a graphql query else { result: false }
+    */
+    identifyGraphqlRequest: function (dataString) {
+      try {
+        const rawDataObj = _.attempt(JSON.parse, dataString.replace(/\r\n/g, ''));
+        if (rawDataObj && !_.isError(rawDataObj)) {
+          if (!_.has(rawDataObj, 'variables')) {
+            rawDataObj.variables = {};
+          }
+          if (_.keys(rawDataObj).length === 2 && _.has(rawDataObj, 'query') && _.has(rawDataObj, 'variables')) {
```

Comment:
@aman-v-singh When we discussed this, I remember there being 3rd optional key as well. Are we not supporting it?

Reply:
@VShingala The 3rd optional key was `operationName`, we don't have any place to show `operationName` in the Postman App `graphql` editor, but I will add a check for that as well. `operationName` is only required if multiple operations are present in the query. https://graphql.org/learn/serving-over-http/#post-request

Reply:
@aman-v-singh That's fine if we don't support it in postman app, but if we just ignore it here than even valid graphql bodies will be ignored and detected as raw body.

Reply:
@VShingala I have added a check for `operationName`.